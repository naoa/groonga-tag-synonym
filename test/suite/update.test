register commands/tag_synonym

table_create Tags TABLE_PAT_KEY ShortText --default_tokenizer TokenDelimit
column_create Tags synonym COLUMN_SCALAR Tags

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos tags COLUMN_SCALAR Tags

load --table Tags
[
{"_key": "Ruby", "synonym": "Rroonga"}
]

tag_synonym_add --table Memos --column tags

load --table Memos
[
{"_key": "Groonga", "tags": "Groonga"},
{"_key": "Rroonga", "tags": "jRuby"}
]

load --table Memos
[
{"_key": "Rroonga", "tags": "Ruby"}
]

select Memos --output_columns 'tags'
