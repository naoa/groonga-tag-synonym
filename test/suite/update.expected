register commands/tag_synonym
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText --default_tokenizer TokenDelimit
[[0,0.0,0.0],true]
column_create Tags synonym COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos tags COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
load --table Tags
[
{"_key": "Ruby", "synonym": "Rroonga"}
]
[[0,0.0,0.0],1]
tag_synonym_add --table Memos --column tags
[[0,0.0,0.0],[1]]
load --table Memos
[
{"_key": "Groonga", "tags": "Groonga"},
{"_key": "Rroonga", "tags": "jRuby"}
]
[[0,0.0,0.0],2]
load --table Memos
[
{"_key": "Rroonga", "tags": "Ruby"}
]
[[0,0.0,0.0],1]
select Memos --output_columns 'tags'
[[0,0.0,0.0],[[[2],[["tags","Tags"]],["Groonga"],["Rroonga"]]]]
